tests/zfs-tests/tests/functional/cli_root/zpool/zpool_colors.ksh

```diff
@@ -0,0 +1,94 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+# Copyright (c) 2019 Lawrence Livermore National Security, LLC.
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# DESCRIPTION:
+#	Test that zpool status colored output works.
+#
+# STRATEGY:
+# 1. Create a pool with a bunch of errors and force fault one of the vdevs.
+# 2. Look for 'pool:' in bold.
+# 3. Look for 'DEGRADED' in yellow
+# 3. Look for 'FAULTED' in red
+#
+
+verify_runnable "both"
+
+function cleanup
+{
+	zinject -c all
+}
+
+log_onexit cleanup
+
+log_assert "Test colorized zpool status output"
+
+DISK2="$(echo $DISKS | cut -d' ' -f2)"
+DISK3="$(echo $DISKS | cut -d' ' -f3)"
+
+log_must dd if=/dev/urandom of=/$TESTDIR/testfile bs=10M count=1
+
+log_must zpool sync
+
+log_must zpool offline -f $TESTPOOL $DISK3
+log_must wait_for_degraded $TESTPOOL
+log_must zinject -d $DISK2 -e io -T read -f 20 $TESTPOOL
+log_must zinject -d $DISK2 -e io -T write -f 20 $TESTPOOL
+
+
+log_must zpool scrub -w $TESTPOOL
+log_must zinject -c all
+
+
+# Use 'script' to fake zpool status into thinking it's running in a tty.
+# Log the output here in case it's needed for postmortem.
+log_note "$(script --return --quiet -c \
```

Comment:
The `script` command will be different on FreeBSD, but it seems to still meet the basic need here. :+1: 
```
$ echo $(script -q /dev/null python2.7 -c "import sys; print sys.stdout.isatty())")
True
```
The discrepancy could be hidden behind a function to make it easier to add alternative implementations.

Reply:
FreeBSD support has landed in ZTS. We don't have FreeBSD bots blocking PRs yet, but let's try to get this addressed now.
I suggest putting the following under the cleanup function:
```
function faketty
{
	if is_freebsd; then
		script -q /dev/null "$@"
	else
		script --return --quiet -c "$*" /dev/null
	fi
}
```
And then use it like so:
```
log_note "$(faketty TERM=xterm-256color ZFS_COLOR=1 zpool status)"
```
```
out="$(faketty TERM=xterm-256color ZFS_COLOR=1 zpool status | \
    grep -E 'pool:|DEGRADED' | \
    sed -r 's/\s+//g;'$(echo -e 's/\033/ESC/g'))"
```
And so on.

Reply:
Thanks for the code.  I'll add it in